pkg/controller/volume/expand/expand_controller.go

```diff
@@ -302,19 +305,31 @@ func (expc *expandController) expand(pvc *v1.PersistentVolumeClaim, pv *v1.Persi
 		return util.DeleteAnnPreResizeCapacity(pv, expc.GetKubeClient())
 	}
 
-	pvc, err := util.MarkResizeInProgressWithResizer(pvc, resizerName, expc.kubeClient)
-	if err != nil {
-		klog.V(5).Infof("Error setting PVC %s in progress with error : %v", util.GetPersistentVolumeClaimQualifiedName(pvc), err)
-		return err
-	}
+	var generatedOptions volumetypes.GeneratedOperations
```

Comment:
(this line is unrelated, just starting a new thread)

Controller's `syncHandler` ignores PVCs that user shrinks back to the original size:
https://github.com/kubernetes/kubernetes/blob/4960d0976ac9b0b663b1825d97eb19f8f0af57ff/pkg/controller/volume/expand/expand_controller.go#L244

(I think `AnnPreResizeCapacity` is either added later or not used at all when the feature gate is enabled).

It should either check the resize status or `allocatedResources` to finish resize in progress.

Reply:
Nevermind, we don't support shrink to the original size.